src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs

```diff
@@ -112,7 +120,7 @@ internal bool AdoptOrphanedBatches()
 
                 AdoptBatches(session, orphanedBatchIds);
 
-                var moreToDo = stats.IsStale || orphanedBatchIds.Any();
+                var moreToDo = abort || stats.IsStale || orphanedBatchIds.Any();
```

Comment:
This doesn't read well. There's more to do if we're aborting. 

The bool value returned is "all done". Perhaps we should add a second bool so we can have:

```CSharp
var allDone = abort || !moreToDo;
return allDone;
```